Sources/MapboxCoreNavigation/CoreFeedbackEvent.swift

```diff
@@ -14,38 +14,33 @@ class CoreFeedbackEvent: Hashable {
     }
     
     func hash(into hasher: inout Hasher) {
-        hasher.combine(id.hashValue)
+        hasher.combine(identifier.hashValue)
     }
     
     static func ==(lhs: CoreFeedbackEvent, rhs: CoreFeedbackEvent) -> Bool {
-        return lhs.id == rhs.id
+        return lhs.identifier == rhs.identifier
     }
-}
-
-class FeedbackEvent: CoreFeedbackEvent {
     
-    convenience init(eventDetails: NavigationEventDetails) {
-        let eventDictionary = (try? eventDetails.asDictionary()) ?? [:]
-        self.init(timestamp: Date(), eventDictionary: eventDictionary)
+    private enum CodingKeys: String, CodingKey {
+        case id
+        case timestamp
+        case eventDictionaryData
     }
     
-    func update(type: FeedbackType, source: FeedbackSource, description: String?) {
-        eventDictionary["feedbackType"] = type.description
-
-        // if there is a subtype for this event then append the subtype description to our list for this type of feedback
-        if let subtypeDescription = type.subtypeDescription {
-            var subtypeList = [String]()
-            if let existingSubtypeList = eventDictionary["feedbackSubType"] as? [String] {
-                subtypeList.append(contentsOf: existingSubtypeList)
-            }
-
-            if !subtypeList.contains(subtypeDescription) {
-                subtypeList.append(subtypeDescription)
-            }
-            eventDictionary["feedbackSubType"] = subtypeList
-        }
-        eventDictionary["source"] = source.description
-        eventDictionary["description"] = description
+    required init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        identifier = try container.decode(UUID.self, forKey: .id)
+        timestamp = try container.decode(Date.self, forKey: .timestamp)
+        let eventDictionaryData = try container.decode(Data.self, forKey: .eventDictionaryData)
+        eventDictionary = try JSONSerialization.jsonObject(with: eventDictionaryData) as? [String: Any] ?? [:]
```

Comment:
Is it expected that the result json isn't `[String: Any]`? If not, then this conversion will silently fail and you will never know about this.

Reply:
I guess there is no way for this function to fail because we user doesn't have access to this field and we get this data from the real JSON object in line 42.

Reply:
Then having an asserting won't hurt. Today everything is ok, tomorrow, something changes in the surrounding code and will start silently failing.